src/moaicore/MOAICCParticleSystem.cpp

```diff
@@ -0,0 +1,2384 @@
+    MOAI_LUA_SETUP ( MOAICCParticleSystem, "U" )
+    
+    self->FlipY();
+    
+    return 0;
+}
+
+//----------------------------------------------------------------//
+/** @name   initializeProperties
+    @text   Sets the emission rate property based on lifespan and maximum particles.  Also sets lifespan term member variable.
+ 
+    @in     MOAICCParticleSystem    self
+    @in     String                  file to load
+    @out    nil
+ */
+int MOAICCParticleSystem::_initializeProperties( lua_State *L ) {
+    MOAI_LUA_SETUP( MOAICCParticleSystem, "U" )
+    
+    self->InitializeEmitter();
+    
+    return 0;
+}
+
+//----------------------------------------------------------------//
+/** @name   load
+    @text   Initialize particle system properties with an XML file.
+ 
+    @in     MOAICCParticleSystem    self
+    @in     String                  file to load
+    @out    nil
+ */
+
+int MOAICCParticleSystem::_load( lua_State *L ) {
+    MOAI_LUA_SETUP( MOAICCParticleSystem, "US" )
+    cc8* xml = lua_tostring( state, 2);
+    
+    if ( MOAILogMessages::CheckFileExists ( xml, L )) {
+        TiXmlDocument doc;
+        doc.LoadFile ( xml );
+        self->ParseXML ( xml, doc.RootElement ());
+        
+    }
+    
+    return 0;
+}
+//----------------------------------------------------------------//
+/** @name   startSystem
+    @text   Begin the generation of particles.  Use in conjenction with start().
+ 
+    @in     MOAICCParticleSystem    self
+    @out    nil
+ */
+
+int MOAICCParticleSystem::_startSystem ( lua_State *L ) {
+    MOAI_LUA_SETUP( MOAICCParticleSystem, "U" )
+    self->StartSystem();
+    return 0;
+}
+
+//----------------------------------------------------------------//
+/** @name   stopSystem
+    @text   Halt the generation of particles.  Any remaining particles will be allowed to stay until the end of their lifespan. Use stop() to halt the animation of particles.
+ 
+    @in     MOAICCParticleSystem    self
+    @out    nil
+ */
+int MOAICCParticleSystem::_stopSystem ( lua_State *L ) {
+    
+    MOAI_LUA_SETUP( MOAICCParticleSystem, "U" )
+    self->StopSystem();
+    
+    return 0;
+}
+//----------------------------------------------------------------//
+/** @name   surge
+    @text   Generate the given number of particles at one time.
+    @in     MOAICCParticleSystem    self
+    @in     number                  numberOfParticles
+    @out    nil
+ */
+
+int MOAICCParticleSystem::_surge ( lua_State *L ) {
+    MOAI_LUA_SETUP( MOAICCParticleSystem, "UN" )
+    u32 numberOfParticles = state.GetValue < u32 >(2, 1.0);
+    self->Surge( numberOfParticles );
+    return 0;
+}
+//----------------------------------------------------------------//
+/** @name   reset
+    @text   Make all particles get cleared out at the next frame.
+ 
+    @in     MOAICCParticleSystem    self
+    @opt    boolean                 activate    Default true.  Whether to begin the generation of particles after clearing the particles.
```

Comment:
I don't think this param is necessary. Better for reset not to touch the active/inactive state. We can always call `start` or `stop` directly.